### ISSUE TITLE
ItemsRepeater scrolling breaks when wrapping in StackPanel

### ISSUE THREAD

**Issue by grokys** [AvaloniaUI/Avalonia#4569]
Repro: in ControlCatalog `ItemsRepeaterPage`, change `ItemsRepeater` to read:

```xml
        <ItemsRepeater Name="repeater" Background="Transparent" Items="{Binding Items}">
          <ItemsRepeater.ItemTemplate>
            <DataTemplate>
              <StackPanel>
                <TextBlock Text="{Binding Text}"/>
              </StackPanel>
            </DataTemplate>
          </ItemsRepeater.ItemTemplate>
        </ItemsRepeater>
```

Scrolling breaks. Without the `StackPanel` everything is fine.

**Comment by FoggyFinder** [comment]
Can it be related to #4453 ?

**Comment by grokys** [comment]
This seems to be caused by a difference in the way layout invalidation is done between UWP (where `ItemsRepeater` is ported from) and WPF (whose layout semantics we broadly follow). Seems that in UWP invalidating a control causes an immediate invalidation of parent controls, whereas in WPF the invalidation only happens when the child control reports a different desired size.

**Comment by grokys** [comment]
Fixed by #10864 and #11127.
